### PR TITLE
Add unit tests for Shell::parseInput and enable Linux compilation

### PR DESCRIPTION
**Pull request by Chlebnik**
This PR introduces unit testing for the `Shell::parseInput` method. 
It includes a new test suite in `tests/TestShell.cpp` and a mock Kernel implementation in `tests/MockKernel.cpp`.
To enable running these tests on Linux, `src/stdafx.h` and other files were modified to conditionally include Windows headers and provide compatible type definitions for Linux.
A bug was also fixed in `src/Struct.h` where `process_data` members were not initialized, causing garbage values in tests.

**Coverage:**
- Simple commands
- Commands with arguments (quoted and unquoted)
- Input/Output redirection
- Pipes
- Edge cases (empty input, invalid syntax)

**Result:**
The `Shell::parseInput` logic is now verified with 12 test cases covering standard usage and edge cases.
The project can now be compiled and tested on Linux environments using `g++`.

---
*PR created automatically by Jules for task [11234361072471535882](https://jules.google.com/task/11234361072471535882) started by @Chlebnik*

### CODE REVIEW

File: src/stdafx.h [hunk 1]
```diff
@@ -6,7 +6,9 @@
 #pragma once
 
 #include <stdio.h>
+#ifdef _WIN32
```
Comment (Chlebnik) [inline src/stdafx.h]: @jules please have only one region branch for same condition

File: src/Utils.cpp [hunk 2]
```diff
@@ -2,21 +2,29 @@
...
 string Utils::WcharToString(wchar_t* text) {
+#ifdef _WIN32
 	string new_text;
 	char ch[260];
 	char DefChar = ' ';
 	WideCharToMultiByte(CP_ACP, 0, text, -1, ch, 260, &DefChar, NULL);
 	new_text = ch;
 
 	return new_text;
+#else
```
Comment (Chlebnik) [inline src/Utils.cpp:13]: There is not functionality for wide chars on unix?
